kai/reactive_codeplanner/task_runner/compiler/maven_validator.py

```diff
@@ -142,7 +143,8 @@ def run_maven(source_directory: Path = Path(".")) -> tuple[int, str, Optional[Pa
     Runs 'mvn compile' and returns the combined stdout and stderr output.
     Also returns the path to the pom.xml file.
     """
-    cmd = ["mvn", "compile"]
+    mavenPath = which("mvn") or "mvn"
```

Comment:
Why use `which`? If it's a windows thing, should we change [our vfs](https://github.com/konveyor/kai/blob/4b6f35fcffd9797278807effe15b8dfeab81d6db/kai/reactive_codeplanner/vfs/git_vfs.py#L99) to use `which` too?

Reply:
git binary on windows isn't a cmd, bat or exe, but maven is. And just doing `mvn` doesn't work in Popen, `git` is just fine. maybe redundant